### PR TITLE
Implements the "setting one setting sets another setting" feature.

### PR DESCRIPTION
**Pull request by colinta**
Adds `setsAnother` property, updates settings before sending the request.

### CODE REVIEW

File: Sources/Controllers/Settings/DynamicSettingCategoryViewController.swift [hunk 1]
```diff
@@ -73,25 +73,42 @@ class DynamicSettingCategoryViewController: UIViewController, UITableViewDataSou
 extension DynamicSettingCategoryViewController: DynamicSettingCellDelegate {
     func toggleSetting(setting: DynamicSetting, value: Bool) {
         if let nav = self.navigationController as? ElloNavigationController {
-            var visibility: [(NSIndexPath, Bool)] = []
+            var visibility: [(NSIndexPath, value: Bool, isVisible: Bool)] = []
```
Comment (steam) [inline Sources/Controllers/Settings/DynamicSettingCategoryViewController.swift]: Now that $1 and $2 have names we should probably name $0 for consistency.

Comment (colinta) [inline Sources/Controllers/Settings/DynamicSettingCategoryViewController.swift]: 👍

**Comment by colinta** [comment]
Specs are green!  Looks like your "off the hip" coding was a success. 😉

**Comment by steam** [comment]
👍

File: Sources/Controllers/Settings/DynamicSettingCategoryViewController.swift [hunk 2]
```diff
@@ -114,9 +116,11 @@ extension DynamicSettingCategoryViewController: DynamicSettingCellDelegate {
...
     }
 
-    func settingChanged(config: SettingConfig, user: User) -> Bool {
-        return DynamicSettingCellPresenter.isVisible(setting: config.setting, currentUser: user) ||
-            user.propertyForSettingsKey(config.setting.key) != config.value
+    private func settingChanged(config: SettingConfig, user: User) -> Bool {
+        let setting = config.setting
+        let currVisibility = DynamicSettingCellPresenter.isVisible(setting: setting, currentUser: user)
+        let currValue = user.propertyForSettingsKey(setting.key)
+        return config.isVisible != currVisibility || config.value != currValue
```
Comment (steam) [inline Sources/Controllers/Settings/DynamicSettingCategoryViewController.swift]: 👍

**Pull request merged**